app/controllers/monthly_reports_controller.rb

```diff
@@ -16,7 +16,7 @@ def show
   end
 
   def new
-    target_month = params[:target_month] || Date.current.last_month.beginning_of_month
+    target_month = params[:target_month] || Date.current.last_month.since(5.days).beginning_of_month
```

Comment:
```
def report_registrable_to
    Time.current.last_month.since(5.days).beginning_of_month
  end
```
にしてpublicにしてみたのですが、
User#report_registrable_monthsの以下部分でうまく動かないことがわかりました。
```
    while month <= report_registrable_to
```
原因は、monthの方は、Dateクラスのインスタンスで、
report_registrable_toは、TimeWithZoneのインスタンスとなっており、
型が違うものを比べているせいっぽいです。

上記以外にもrspecが5箇所失敗しています。
`report_registrable_to`の修正をすると今回のISSUEと関係のない修正がどんどん入ってきそうなのですが、
このISSUEでやっちゃてもいいですか？
それとも別ISSUEでやった方がいいですか？